### PR TITLE
Move localization from directory-based to i18n-based

### PR DESCRIPTION
**Pull request by FerPerales**
This commit removes the need of having an specific folder per language
inside /docs

With only one top-level doc, we can handle translations on their
corresponding files now living under `locales`

Each language now has its own folder, which is going to make easier to
implement continuous localization with Crowdin

A quick way to test this change is:

* Run app locally with `bundle exec rake run`
* Go to `localhost:9292/installfest`. It should be in English by default
* Add `?locale=es` to `localhost:9292/installfest`
* Page will be in Spanish now using translated keys from yml files
* Replace `?locale=es` with `?locale=zh-tw`
* Page with be in Chinese now

See gif below:
locales

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -57,36 +54,36 @@ class InstallFest < Sinatra::Application   # todo: use Sinatra::Base instead, wi
...
-    host.split(".").first
+    host.split('.').first
   end
 
   def default_site
     if host && sites.include?(site = subdomain)
       site
     else
-      DEFAULT_SITES[I18n.locale.to_sym] # no symbol DoS because it's whitelisted
+      'docs'
```
Comment (mxie) [inline app.rb:73]: This might be a good candidate as a constant instead.

File: app.rb [hunk 2]
```diff
@@ -107,24 +104,20 @@ def redirect_pages
...
 
   before do
-    begin
-      I18n.locale = dynamic_locale
-    rescue I18n::InvalidLocale
-      I18n.locale = I18n.default_locale
-    end
+    I18n.locale = dynamic_locale
+  rescue I18n::InvalidLocale
+    I18n.locale = I18n.default_locale
```
Comment (mxie) [inline app.rb:109]: Nice 👍

File: app.rb [hunk 3]
```diff
@@ -107,24 +104,20 @@ def redirect_pages
...
   end
 
   after '/:site/*' do
     # Any real page (starts with a site and doesn't end with an extension)
     # gets saved as the 'back' for the next pageload.
-    if sites.include?(params[:site]) && !request.fullpath.match(/\..+\z/)
-      cookies[:docs_back_path] = request.fullpath
-    end
+    cookies[:docs_back_path] = request.fullpath if sites.include?(params[:site]) && !request.fullpath.match(/\..+\z/)
```
Comment (mxie) [inline app.rb:115]: Since the conditions are on the longer side, I'd rather we stick to the block format to ensure readability.

File: app.rb [hunk 4]
```diff
@@ -107,24 +104,20 @@ def redirect_pages
...
   end
 
   def dynamic_locale
-    (params && (params[:locale] or params[:l])) or
-      (host && AVAILABLE_LOCALES.include?(subdomain) && subdomain) or
+    (params && (params[:locale] || params[:l])) ||
+      (host && AVAILABLE_LOCALES.include?(subdomain) && subdomain) ||
```
Comment (mxie) [inline app.rb:120]: Good catch!

File: spec/site_spec.rb [hunk 5]
```diff
@@ -9,18 +9,14 @@
...
   end
-  
+
   it "finds the sites_dir" do
     Site.sites_dir.should == File.expand_path(File.join(File.dirname(__FILE__), "..", "sites", "en"))
   end
-
-  it "finds the sites_dir for Spanish locale" do
-    Site.sites_dir('es').should == File.expand_path(File.join(File.dirname(__FILE__), "..", "sites", "es"))
-  end
```
Comment (mxie) [inline spec/site_spec.rb:25]: Shouldn't have a test _somewhere_ to ensure that the params worked correctly?

**Review by mxie — COMMENTED** [review]
Thanks for working on this!! Left a few comments. Let me know if you have any questions. Seems straightforward to me so far.

File: spec/site_spec.rb (hunk 5 above)
Comment (FerPerales) [reply spec/site_spec.rb:25]: @mxie definitively! adding one

File: app.rb [hunk 6]
```diff
@@ -9,45 +9,42 @@
...
-require "site"
+require 'doc_page'
+require 'step_page'
+require 'markdown_page'
+require 'media_wiki_page'
+require 'raw_page'
+require 'deck'
+require 'deck/rack_app'
+require 'titleizer'
+require 'site'
```
[the remark this answers is not in the document]
Comment (FerPerales) [reply app.rb:24]: @mxie Agree. I'll extract those changes into a different PR

**Review by mxie — APPROVED** [review]
LGTM! Thank you for doing this work! Sorry it took so long for me to review.

**Pull request merged**